### PR TITLE
Chapter 7: Inheritance

### PR DESCRIPTION
**Pull request by speudusa**

### CODE REVIEW

File: content/inheritance/reading/csharp-inheritance/_index.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+   The shaded portion of these boxes can include additional information about each class. We’ll learn about what we might put here in a little bit.
+
+Inheritance is an essential mechanism for sharing data and behavior between related classes. Using it effectively creates organized code with groups of classes that have increasingly specialized behavior.
+
+When this happens, we can visualize the inheritance structure with a slightly more complex diagram.
+
+   {{< rawhtml >}}
+   <img src="../pictures/inheritance-tree.png" alt="Inheritance tree with many nodes" width=80% />
+   {{< /rawhtml >}}
```
Comment (codinglikeagirl42) [inline content/inheritance/reading/csharp-inheritance/_index.md:47]: Picture is not showing -> 
Missing Photo

Picture folder is at the wrong level, problem may be solved by moving the picture folder into the reading folder.

File: content/inheritance/reading/_index.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+draft: false
+weight: 1
+originalAuthor: Courtney Frey # to be set by page creator
+originalAuthorGitHub: speudusa # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
```
Comment (codinglikeagirl42) [inline content/inheritance/reading/_index.md:13]: There are no links.

File: content/inheritance/reading/cats-inheritance/_index.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+originalAuthorGitHub: speudusa # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+<!-- TODO: Link to repository -->
+
+Let’s revisit our `Cat` and `HouseCat` friends. Check out this [repository](LINK) for the code for the lesson. In Visual Studio, open the solution and examine the two classes inside the Cats project.
```
Comment (codinglikeagirl42) [inline content/inheritance/reading/cats-inheritance/_index.md:16]: Code link?

File: content/inheritance/reading/class-diagrams/_index.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+   1. `Student`: Represents an individual student
+   1. `Transcript`: Represents a student’s transcript, including classes that they have taken
+   1. `CourseRecord`: Represents an individual course that a student has completed, along with the grade achieved.
+
+We can represent these three classes and their relationships to each other in the following diagram:
+
+   {{< rawhtml >}}
+   <img src="../pictures/relationships.png" alt="Relationship class diagram" width=80% />
+   {{< /rawhtml >}}
```
Comment (codinglikeagirl42) [inline content/inheritance/reading/class-diagrams/_index.md:53]: Picture missing, should be resolved by moving the picture folder into the reading folder.

File: content/inheritance/exercises/_index.md [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+
+Before you start coding anything inside these classes, diagram how the three classes are going to be related to each other.  You can start with the diagram below.  Be sure to show how the base class extends each subclass.
+
+{{< mermaid >}}
+classDiagram
+  Computer -- Smartphone
+  Computer -- Laptop
+{{< /mermaid >}}
+
+{{% expand "Check your solution" %}}
```
Comment (codinglikeagirl42) [inline content/inheritance/exercises/_index.md:29]: Please tab the Check your solution to match the location of the ones in the ordered list below.

File: content/inheritance/exercises/_index.md [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+      // code continues
+      
+   ```
+   {{% /expand %}}
+
+## Class implementation
+Implement your design and add unit tests to a new TechnologyTests MSTest project.
+
+Add a new MSTest project to your solution.
+{{% expand "Check your solution" %}}
```
Comment (codinglikeagirl42) [inline content/inheritance/exercises/_index.md:152]: Please tab these in two match the location of the ordered list Check your solution.

File: content/inheritance/exercises/_index.md [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+         Id = nextId;
+         nextId++;
+      }
+   }
+   ```
+   {{% /expand %}}
+
+1. Add this class to your `Technology` project above, and add `AbstractEntity` to the class hierarchy in the correct place.
+
+{{% expand "Check your solution" %}}
```
Comment (codinglikeagirl42) [inline content/inheritance/exercises/_index.md:224]: Please tab these in to match the above location, so they are nested under the ordered list.

**Review by codinglikeagirl42 — CHANGES_REQUESTED** [review]
Please update the metadata for all the pages.  The reading subfolder files need to be changed from _index.md to index.md. 

InheritanceExDiagram.png doesn't seem to be being used -> should it be in the exercises?  If so will need to make a separate picture folder.

Please let me know if you have any questions.

File: content/inheritance/_index.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+## Major Concepts 
+
+1. Understand what inheritance is and how programmers can use it.
```
Comment (codinglikeagirl42) [inline content/inheritance/_index.md:17]:
```suggestion
## Major Concepts 

After completing this chapter, you should be able to do the following:

1. Understand what inheritance is and how programmers can use it.
```

File: content/inheritance/reading/csharp-inheritance/_index.md (hunk 1 above)
Comment (speudusa) [reply content/inheritance/reading/csharp-inheritance/_index.md:47]: fixed. thanks for catching this!

File: content/inheritance/exercises/_index.md (hunk 7 above)
Comment (speudusa) [reply content/inheritance/exercises/_index.md:224]: done.

File: content/inheritance/exercises/_index.md (hunk 6 above)
Comment (speudusa) [reply content/inheritance/exercises/_index.md:152]: done.

File: content/inheritance/exercises/_index.md (hunk 5 above)
Comment (speudusa) [reply content/inheritance/exercises/_index.md:29]: done.

File: content/inheritance/reading/class-diagrams/_index.md (hunk 4 above)
Comment (speudusa) [reply content/inheritance/reading/class-diagrams/_index.md:53]: fixed this. needed one more'../'

File: content/inheritance/reading/_index.md (hunk 2 above)
Comment (speudusa) [reply content/inheritance/reading/_index.md:13]: fixed. added key terms and concepts too.

File: content/inheritance/reading/cats-inheritance/_index.md (hunk 3 above)
Comment (speudusa) [reply content/inheritance/reading/cats-inheritance/_index.md]: rephrased and redirected students to the examples repo. thanks!

**Comment by speudusa** [comment]
metadata and files updated.  removing the InheritanceExDiagram from the project. i used the mermaid diagram feature instead.

File: content/inheritance/reading/_index.md (hunk 2 above)
Comment (codinglikeagirl42) [reply content/inheritance/reading/_index.md]: Thanks.

**Review by codinglikeagirl42 — APPROVED** [review]
One minor change on the Reading _index.md page. otherwise looks good, merge away!

File: content/inheritance/reading/_index.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+- `abstract`, abstract class, abstract methods
+
+### Casting
+- runtime exception, polymorphism
+
+### Class Diagrams
+- class diagrams, one-to-one relationship, one-to-many relationship
+
+## Content Links
+
```
[the remark this answers is not in the document]
Comment (codinglikeagirl42) [reply content/inheritance/reading/_index.md:46]: Reading Page just has links.  This information was also on the main Inheritance page.

[the remark this answers is not in the document]
Comment (speudusa) [reply content/inheritance/reading/_index.md]: done. :)

**Pull request merged**